pkg/pillar/zedbox/zedbox.go

```diff
@@ -316,3 +332,56 @@ func startAgentAndDone(agentName string, srvPs *pubsub.PubSub,
 		log.Fatalf("Error write done file: %v", err)
 	}
 }
+
+const maxsize = 2048
+
+// Use a buffer pool to minimize memory usage
+// While this isn't critical here, doing the same as in pubsub/socketdriver
+// makes it easier to test things using zedbox publish and subscribe
+var bufPool = &sync.Pool{
+	New: func() interface{} {
+		buffer := make([]byte, maxsize+1)
+		return buffer
+	},
+}
+
+//getBuffer returns a buffer and a done func to call at defer.
+func getBuffer() ([]byte, func()) {
+	buf := bufPool.Get().([]byte)
+	return buf, func() {
+		bufPool.Put(buf)
+	}
+}
+
+// check and waits till conn's fd is readable
+func connReadCheck(conn net.Conn) error {
```

Comment:
Can we put this method in some common place so that both zedbox and socketdriver and call it? In that way we can avoid duplication.

Reply:
Sure I'll find a place. But shouldn't we move the zedbox reverse pubsub to the pubsub package as part of that? Let me take a look tomorrow.

Reply:
@adarsh-zededa the 4th commit has the refactoring/moving into pubsub/reverse package
If we use this pattern (where the subscriber listens and publisher connects) in more places it might make sense to make it an option in the normal NewSubscriber/NewPublisher, but let's defer that for now.